api/server.go

```diff
@@ -0,0 +1,50 @@
+package api
+
+import (
+	"time"
+
+	"github.com/labstack/echo/v4"
+	"github.com/labstack/echo/v4/middleware"
+)
+
+type Config struct {
+	Addr string
+}
+
+type Server struct {
+	config Config
+	router *echo.Echo
+}
+
+func NewServer(config Config) *Server {
+	server := &Server{
+		config: config,
+	}
+
+	server.setupRouter()
+
+	return server
+}
+
+func (s *Server) setupRouter() {
+	e := echo.New()
+
+	// Server settings
+	e.Server.ReadTimeout = 10 * time.Second
```

Comment:
Can we store this config value into some sort of properties file? so it is configurable there (and later on, we won't be needing code changes here when we need to tweak the values).